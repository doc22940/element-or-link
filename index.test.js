import React from 'react';
import { render } from '@testing-library/react';
import { StaticRouter } from 'react-router-dom';

import { ElementOrLink } from './';

describe('<ElementOrLink />', () => {
  it('renders a div', () => {
    const { container } = render(<ElementOrLink>Hey!</ElementOrLink>);
    expect(container).toMatchSnapshot();
  });

  it('renders a span', () => {
    const { container } = render(
      <ElementOrLink element="span">Hey!</ElementOrLink>
    );
    expect(container).toMatchSnapshot();
  });

  it('renders a link', () => {
    const { container } = render(
      <StaticRouter>
        <ElementOrLink to="/hello">Hey!</ElementOrLink>
      </StaticRouter>
    );
    expect(container).toMatchSnapshot();
  });

  it('renders a div even if a link is provided', () => {
    const { container } = render(
      <StaticRouter>
        <ElementOrLink to="/hello" displayElement={true}>
          Hey!
        </ElementOrLink>
      </StaticRouter>
    );
    expect(container).toMatchSnapshot();
  });

  it('renders a div if link is an empty string', () => {
    const { container } = render(
      <StaticRouter>
        <ElementOrLink to="">Hey!</ElementOrLink>
      </StaticRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
