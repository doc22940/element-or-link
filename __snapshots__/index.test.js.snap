// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ElementOrLink /> renders a div 1`] = `
<div>
  <div>
    Hey!
  </div>
</div>
`;

exports[`<ElementOrLink /> renders a link 1`] = `
<div>
  <a
    href="/hello"
  >
    Hey!
  </a>
</div>
`;

exports[`<ElementOrLink /> renders a span 1`] = `
<div>
  <span>
    Hey!
  </span>
</div>
`;

exports[`<ElementOrLink /> renders a span even if a link is provided 1`] = `
<div>
  <span>
    Hey!
  </span>
</div>
`;

exports[`<ElementOrLink /> renders a span if link is an empty string 1`] = `
<div>
  <span>
    Hey!
  </span>
</div>
`;
